### PR TITLE
Halligan Improvements

### LINKED ISSUE #33516: Halligan issues
(WEB REPORT BY: poales REMOTE: 206.221.180.138:7777)
# Revision
3a65018cd7c03a75cf34870b2e664d9bd567ced4

# Description

1. You can use the halligan on doors that have been hacked open, and it will try to open them again
2. If you halligan a door with no power, it still says it's struggling with the motors, like using a crowbar on a powered airlock. It really should just open it like a normal crowbar

### PR DESCRIPTION
**Pull request by west3436**
## What this does

A number of improvements to the Halligan bar:
1. Halligan bar can now be used to cut open closed shutters. 30 seconds to cut the hole and 5 seconds to pry the hole open (same duration as breaking a bolted door open). Damaged shutters can be fixed with a welding tool.
2. Now defaults to standard crowbar behavior on unpowered airlocks rather than taking the full duration to open them.
3. You will no longer attempt to pry open already-opened doors (closes #33516)
4. Set of Irons can now be stored in the atmos tech and CE suits.
5. Halligan bar now spawns in the CE locker in addition to the atmos tech lockers.

## Why it's good

Helps make the Halligan bar more consistent as a crowbar substitute, gives one to the CE (whom should have had it from the start), allows the Set of Irons to be carried in more suits than just the fire suit, and gives a little bit more functionality to it since it currently is much less efficient than just hacking doors open.

## Changelog

:cl:
 * rscadd: Halligan bars can now be used to cut a hole through shutters.
 * rscadd: A Halligan bar will now spawn in the CE closet.
 * bugfix: Halligan bars will now use the default crowbar behavior on an unpowered airlock.
 * tweak: A Set of Irons can now be stored in atmos tech and CE hardsuit storage slots.

### CODE REVIEW

**Comment by west3436** [comment]
Very open to a new sprite for the hole in shutters - I am not a spriter by any means.

**Comment by west3436** [comment]
need2bugfix

**Comment by Hinaichigo** [comment]
Ghetto halligan wen.

**Comment by west3436** [comment]
> Ghetto halligan wen.

They can already be ordered in advanced firefighting crates

**Comment by ManopolisJones** [comment]
Why do they cut holes in shutters and not just pop then open? I mean, shouldn't the axe be putting holes in things?

**Comment by west3436** [comment]
> Why do they cut holes in shutters and not just pop then open? I mean, shouldn't the axe be putting holes in things?

Halligan bars are used irl to pierce and pry things open like garage doors

**Comment by githubuser4141** [comment]
I think this'd affect 'map balance' since there's a lot of shutters used in place of blast doors for aesthetical purposes (E.g. In the Armoury, Bridge, Captain's Office, etc.) And until now there wasn't any difference between blast doors/shutters mechanics wise. (?)

**Comment by west3436** [comment]
> I think this'd affect 'map balance' since there's a lot of shutters used in place of blast doors for aesthetical purposes (E.g. In the Armoury, Bridge, Captain's Office, etc.) And until now there wasn't any difference between blast doors/shutters mechanics wise. (?)

If we run into cases of balance issues I'll be happy to map in blast doors where needed. Keep in mind that these aren't super accessible already.

File: code/game/machinery/doors/shutters.dm [hunk 1]
```diff
@@ -6,30 +6,80 @@
...
-			operating = 0
+	if(istype(I,/obj/item/tool/crowbar/halligan))
+		var/obj/item/tool/crowbar/halligan/H = I
+		if(density) //can't cut an open door open, dumbass
+			if(cut_open)
+				if(!pried_open)
+					pry(user)
+					return
+			else
+				if(!pried_open)
```
Comment (Kurfursten) [inline code/game/machinery/doors/shutters.dm]: Both branches check pried open, move this up to the density check

**Comment by Blithering** [comment]
> > Why do they cut holes in shutters and not just pop then open? I mean, shouldn't the axe be putting holes in things?
> 
> Halligan bars are used irl to pierce and pry things open like garage doors

Can the shutters be repaired once they've been can openered? they're not buildable as far as I know so you could do some permanent damage by getting your hands on one.

**Comment by west3436** [comment]
> > > Why do they cut holes in shutters and not just pop then open? I mean, shouldn't the axe be putting holes in things?
> > 
> > 
> > Halligan bars are used irl to pierce and pry things open like garage doors
> 
> Can the shutters be repaired once they've been can openered? they're not buildable as far as I know so you could do some permanent damage by getting your hands on one.

Yeah, if it's been fully opened you add some metal sheets then weld it back together. If only the first step has been completed you only need to weld it.

File: code/game/machinery/doors/shutters.dm (hunk 1 above)
Comment (west3436) [reply code/game/machinery/doors/shutters.dm]: resolved in latest commit

File: code/game/machinery/doors/shutters.dm [hunk 2]
```diff
@@ -19,15 +19,15 @@
 	add_fingerprint(user)
 	if(istype(I,/obj/item/tool/crowbar/halligan))
 		var/obj/item/tool/crowbar/halligan/H = I
-		if(density) //can't cut an open door open, dumbass
+		if(density) //can't cut an open door open
+			if(pried_open) //shutter is already fully cut open
```
Comment (Kurfursten) [inline code/game/machinery/doors/shutters.dm]: there is no reason for this to be its own if statement instead of density && !pried_open

Comment (west3436) [reply code/game/machinery/doors/shutters.dm]: right

**Comment by pedr69** [comment]
hm, this only works on shutters and not the blast doors
still need a mech drill to get through those

**Pull request merged**